src/app/components/popup-host/popup-host.component.ts

```diff
@@ -0,0 +1,23 @@
+import {Component, Input, TemplateRef, ViewChild, ViewContainerRef} from '@angular/core';
+
+@Component({
+    selector: 'app-popup-host',
+    templateUrl: './popup-host.component.html',
+    styleUrls: ['./popup-host.component.css'],
+})
+export class PopupHostComponent {
+    @ViewChild('viewport', {static: true, read: ViewContainerRef})
+    viewport: ViewContainerRef | undefined;
+
+    hidden = true;
+
+    @Input() set template(template: TemplateRef<unknown> | null) {
+        if (template) {
+            this.viewport?.clear();
+            this.viewport?.createEmbeddedView(template);
+            this.hidden = false;
+        } else {
+            this.hidden = true;
+        }
```

Comment:
Рекомендую вынести общую часть за пределы if дабы уйти от копипасты
```suggestion
        this.hidden = !template;

        if (template) {
            this.viewport?.clear();
            this.viewport?.createEmbeddedView(template);
        }
```

Вижу, что по твоей логике не удаляется view при получении `null` в инпуте. Почему решил так реализовать, нет ли подводных камней связанных с Incremential DOM?)